### DISCUSSION TITLE
Question: How to integrate customized BackgroundTasks implementation?

### DISCUSSION THREAD

**Discussion by savery-endpoint** [fastapi/fastapi#8478] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import time
import typing

from fastapi import FastAPI
from starlette.background import BackgroundTasks, BackgroundTask


app = FastAPI()


class MyBGTasks(BackgroundTasks):
    def add_task(self, func: typing.Callable, *args, **kwargs) -> None:
        print("Wow, you added a new task :)")
        task = BackgroundTask(func, *args, **kwargs)
        self.tasks.append(task)


def task_fn():
    print("Doing a task yay")
    time.sleep(2)
    print("Done")


@app.get("/")
def route_handler(background_tasks: MyBGTasks):
    background_tasks.add_task(task_fn)
    return {"hello": "world"}
```

### Description

Why is my custom BackgroundTasks class not loaded here? Is it possible to use custom background tasks implementation? Why does `def route_handler(background_tasks: MyBGTasks = Depends(MyBGTasks)):` not work?

### Operating System

macOS

### FastAPI Version

0.75.2

### Python Version

Python 3.10.1

**Comment by Kludex** [comment, marked as answer]
> Is it possible to use custom background tasks implementation?

No.

> Why does def route_handler(background_tasks: MyBGTasks = Depends(MyBGTasks)): not work?

Because only `BackgroundTasks` can be used there.

**Comment by xingdongzhe** [comment]
@Kludex So can we use the user annotation(Here is `MyBGTasks `) to get  subclass of `BackgroundTasks ` instance?
I deep down into `fastapi.dependencies.utils.get_dependant` and it uses the `lenient_issubclass(param.annotation, BackgroundTasks)`  to get `background_tasks_param_name`.
